scripts/west_commands/runners/qemu.py

```diff
@@ -4,28 +4,210 @@
+                            help='GDB server, default is JLinkGDBServer')
+        parser.add_argument('--gdb-host', default='',
+                            help='custom gdb host, defaults to the empty string '
+                            'and runs a gdb server')
+        parser.add_argument('--gdb-port', default=DEFAULT_QEMU_GDB_PORT,
+                            help='pyocd gdb port, defaults to {}'.format(
+                                DEFAULT_QEMU_GDB_PORT))
+        parser.add_argument('--dtb', required=False, dest='dtb',
+                            help='specifies a dtb file')
+        parser.add_argument('--qemu_option', required=False, dest='qemu_opts',
+                            help='specifies qemu opts')
+        parser.add_argument('--commander', default=DEFAULT_QEMU_EXE,
+                            help=f'''QEMU executable, default is
+                            {DEFAULT_QEMU_EXE}''')
+        parser.add_argument('--pipe', default=DEFAULT_QEMU_FIFO,
+                            help=f'''if given, qemu use pipe''')
 
     @classmethod
     def do_create(cls, cfg, args):
-        return QemuBinaryRunner(cfg)
+        return QemuBinaryRunner(cfg, machine=args.machine,
+                                 cpu=args.cpu,
+                                 dtb=args.dtb,
+                                 qemu_opts=args.qemu_opts,
+                                 commander=args.commander,
+                                 pipe=args.pipe,
+                                 gdbserver=args.gdbserver,
+                                 gdb_host=args.gdb_host,
+                                 gdb_port=args.gdb_port,
+                                 tui=args.tui, tool_opt=args.tool_opt)
 
     def do_run(self, command, **kwargs):
-        pass
+        self.commander = os.fspath(
+            Path(self.require(self.commander)).resolve())
+
+        #server_cmd = ([self.gdbserver] +
+        #              ['-port', str(self.gdb_port)] + self.tool_opt)
+
+        if command == 'flash':
+            self.flash(**kwargs)
+        elif command == 'debug':
+            self.debug(**kwargs)
+        else:
+            self.logger.warnning("%s not supported yet", format(command))
+
+    def get_qemu_opts(self):
+
+        def fstr(template):
+            return eval(f"f'{template}'", {'self' : self})
+        if self.qemu_opts:
+            with open(self.qemu_opts, 'r') as stream:
+                try:
+                    _d = yaml.safe_load(stream)
+                    if "qemu" in _d and "qemu_option" in _d["qemu"]:
+                        for _li in _d["qemu"]["qemu_option"]:
+                            li_parsed = fstr(_li).format()
+                            if "loader,file" in _li:
+                                self.use_loader = True
+                            self.qemu_opts_content += li_parsed.split(" ")
+                except yaml.YAMLError as _e:
+                    print(_e)
```

Comment:
What's going on here; why isn't this fatal? Why is this using `print`?

Reply:
let me fix it